### PR TITLE
Qt4 Retirement Survey (January 2024)

### PR DESCRIPTION
**Pull request by jiegec**
Topic Description
-----------------

- astrometry.net: update to 0.94

- libnova: fix missing debug symbols

- scantailor: make transitional package for scantailor-advanced

- scantailor-advanced: new, 1.0.19

- suil: update to 0.10.20

- lv2: update to 1.18.10
    Remove legacy symlinks under /usr/include/lv2
- kstars: update to 3.6.7
    - QJson is no longer required
    - Remove legacy patch
- libindi: update to 2.0.5

- stellarsolver: new, 2.5

- qmmp: update to 1.6.5

... and 21 more commits

Package(s) Affected
-------------------

- libnova: 0.15.0-3
- libindi: 2.0.5
- quazip: 0.9.1-3
- qjson: 1:0.9.0-4
- adwaita-qt: 1.4.1-1
- scantailor-advanced: 1.0.19
- scantailor: 1:0
- lyx: 2.3.7
- scribus: 1.6.1
- kstars: 1:3.6.7
- phonon: 1:4.11.1-5
- poppler: 1:22.08.0-2
- astrometry.net: 0.94
- stellarsolver: 2.5
- qmmp: 1.6.5
- zynaddsubfx: 3.0.6
- carla: 1:2.5.8
- ntk: 1:1.3.1001-1
- libdbusmenu-qt: 1:0.9.3+16.04.20160218-7
- fcitx5-qt: 1:5.1.3-1
- projectm: 4.0.0
- lash: 0.6.0~rc2-2
- libechonest: 2.3.1-7
- qt-gstreamer: 1.2.0-11
- suil: 0.10.20
- lv2: 1.18.10
- liblastfm: 1.0.9-7
- linuxsampler: 2.3.0
- libgig: 4.4.0
- gwenhywfar: 5.4.0-3
- qca: 2.3.4-3

Security Update?
----------------

No

Build Order
-----------

```
#buildit qt-gstreamer liblastfm adwaita-qt poppler qca gwenhywfar scribus projectm libgig linuxsampler ntk lash zynaddsubfx carla quazip phonon fcitx5-qt qjson libdbusmenu-qt libechonest stellarsolver libnova libindi astrometry.net kstars lyx qmmp lv2 suil scantailor-advanced scantailor
```

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`
- [x] AArch64 `arm64`

**Secondary Architectures**

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`
- [x] RISC-V 64-bit `riscv64`

**Experimental Architectures**

- [ ] MIPS R6 64-bit (Little Endian) `mips64r6el`

### CODE REVIEW

**Comment by MingcongBai** [comment]
图片

QMMP seems to have problems recognising CJK-named folders.

**Comment by MingcongBai** [comment]
It's also unable to play any audio files on my computer (including *.mp3).

**Comment by MingcongBai** [comment]
ZynAddSubfx should be using ntk, but it seems to be using fltk.

File: app-scientific/stellarsolver/autobuild/defines [hunk 1]
```diff
@@ -0,0 +1,4 @@
+PKGNAME=stellarsolver
+PKGSEC=math
+PKGDEP="gsl wcslib qt-5"
+PKGDES="Star extractor and astrometric solver"
```
Comment (MingcongBai) [inline app-scientific/stellarsolver/autobuild/defines]: Stellar data extractor and astrometric solver

File: app-utils/testdisk/autobuild/defines [hunk 2]
```diff
@@ -1,6 +1,6 @@
 PKGNAME=testdisk
 PKGSEC=utils
-PKGDEP="libjpeg-turbo libreiserfs ntfs-3g openssl qt-4"
+PKGDEP="libjpeg-turbo libreiserfs ntfs-3g openssl qt-5"
 PKGDES="Checks and undeletes partitions, and utility for photo recovery"
 
 AUTOTOOLS_AFTER="--without-ewf --enable-sudo --enable-qt"
```
Comment (MingcongBai) [inline app-utils/testdisk/autobuild/defines]: The resulting package does not seem to come with a GUI.

File: runtime-desktop/ntk/autobuild/build [hunk 3]
```diff
@@ -1,4 +1,12 @@
+abinfo "Configuring ntk .."
 chmod a+x waf
 ./waf configure --prefix=/usr
```
Comment (MingcongBai) [inline runtime-desktop/ntk/autobuild/build]: Please use absolute directories here.

File: runtime-desktop/ntk/autobuild/build [hunk 4]
```diff
@@ -1,4 +1,12 @@
+abinfo "Configuring ntk .."
 chmod a+x waf
```
Comment (MingcongBai) [inline runtime-desktop/ntk/autobuild/build]: Verbosity and absolute directory here.

File: runtime-desktop/ntk/autobuild/build [hunk 5]
```diff
@@ -1,4 +1,12 @@
...
 ./waf configure --prefix=/usr
+
+abinfo "Building ntk .."
 ./waf -p -j$(nproc)
```
Comment (MingcongBai) [inline runtime-desktop/ntk/autobuild/build]: Ditto.

File: runtime-desktop/ntk/autobuild/build [hunk 6]
```diff
@@ -1,4 +1,12 @@
...
+
+abinfo "Installing ntk .."
 ./waf --destdir="$PKGDIR" install
```
Comment (MingcongBai) [inline runtime-desktop/ntk/autobuild/build]: Ditto.

File: runtime-doc/poppler/autobuild/prepare [hunk 7]
```diff
@@ -1,4 +1 @@
-ln -sv /usr/lib/qt4/bin/moc "$SRCDIR"/moc-qt4
-export PATH="${SRCDIR}:$PATH"
-
 export CXXFLAGS="${CXXFLAGS} -I/usr/include/openjpeg-2.3"
```
Comment (MingcongBai) [inline runtime-doc/poppler/autobuild/prepare]: abinfo here.

File: runtime-multimedia/libechonest/autobuild/defines [hunk 8]
```diff
@@ -1,5 +1,9 @@
 PKGNAME=libechonest
 PKGSEC=libs
-PKGDEP="qjson qt-5"
+PKGDEP="qt-5"
 BUILDDEP="cmake doxygen"
 PKGDES="C++ library for interfacing with Echo Nest"
+
+CMAKE_AFTER="-DECHONEST_BUILD_TESTS=OFF \
+             -DBUILD_WITH_QT4=OFF"
+
```
Comment (MingcongBai) [inline runtime-multimedia/libechonest/autobuild/defines]: Extraneous new line?

File: runtime-multimedia/lv2/autobuild/preinst [hunk 9]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
```
Comment (MingcongBai) [inline runtime-multimedia/lv2/autobuild/preinst]: Shebang is not needed.

File: app-multimedia/qmmp/autobuild/defines [hunk 10]
```diff
@@ -1,6 +1,6 @@
 PKGNAME=qmmp
 PKGSEC=sound
-PKGDEP="curl libmad libogg libvorbis qt-5 taglib xdg-utils faad2 ffmpeg flac libcddb \
+PKGDEP="curl libmad libogg libvorbis qt-6 taglib xdg-utils faad2 ffmpeg flac libcddb \
```
Comment (MingcongBai) [inline app-multimedia/qmmp/autobuild/defines]: Please don't upgrade to Qt 6 yet. We do not have the proper theming engines (provided by KDE) and the program will have issue rendering file dialogues properly.

Please use the 1.6 branch for Qt 5 for now.

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: app-scientific/stellarsolver/autobuild/defines (hunk 1 above)
Comment (jiegec) [reply app-scientific/stellarsolver/autobuild/defines]: Done

File: app-utils/testdisk/autobuild/defines (hunk 2 above)
Comment (jiegec) [reply app-utils/testdisk/autobuild/defines]: testdisk only supports qt4, so we drop this commit.

File: runtime-desktop/ntk/autobuild/build (hunk 3 above)
Comment (jiegec) [reply runtime-desktop/ntk/autobuild/build]: Updated to use `python3 "$SRCDIR/waf"`

File: runtime-desktop/ntk/autobuild/build (hunk 4 above)
Comment (jiegec) [reply runtime-desktop/ntk/autobuild/build]: Updated to use python3 "$SRCDIR/waf"

File: runtime-desktop/ntk/autobuild/build (hunk 5 above)
Comment (jiegec) [reply runtime-desktop/ntk/autobuild/build]: Updated to use `python3 "$SRCDIR/waf"`

File: runtime-desktop/ntk/autobuild/build (hunk 6 above)
Comment (jiegec) [reply runtime-desktop/ntk/autobuild/build]: Updated to use `python3 "$SRCDIR/waf"`

File: runtime-doc/poppler/autobuild/prepare (hunk 7 above)
Comment (jiegec) [reply runtime-doc/poppler/autobuild/prepare]: The extra CXXFLAGS is no longer required: upstream can automatically detect openjpeg.

File: runtime-multimedia/libechonest/autobuild/defines (hunk 8 above)
Comment (jiegec) [reply runtime-multimedia/libechonest/autobuild/defines]: Fixed.

File: runtime-multimedia/lv2/autobuild/preinst (hunk 9 above)
Comment (jiegec) [reply runtime-multimedia/lv2/autobuild/preinst]: Fixed.

File: app-multimedia/qmmp/autobuild/defines (hunk 10 above)
Comment (jiegec) [reply app-multimedia/qmmp/autobuild/defines]: Changed to use qmmp 1.6.5

**Review by MingcongBai — APPROVED** [review]

**Pull request merged**